src/SearchFilter.tsx

```diff
@@ -1,49 +1,132 @@
 import React, { FunctionComponent } from 'react';
+import chroma from 'chroma-js';
+import Select, { ValueType } from 'react-select';
 
 interface SearchFilterProps {
-  handleChange: (event: React.FormEvent<HTMLFormElement>) => void;
+  handleNameChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleTypeChange: (selectedType: ValueType<typeSelectObject>) => void;
 }
 
-const generateTypeCheckbox = () => {
+interface typeSelectObject {
+  value: string;
+  label: string;
+  color: string;
+}
+
+interface optionObject {
+  data: typeSelectObject;
+  isDisabled: boolean;
+  isFocused: boolean;
+  isSelected: boolean;
+}
+
+const SearchFilter: FunctionComponent<SearchFilterProps> = props => {
   const types = [
-    'normal',
-    'fire',
-    'fighting',
-    'water',
-    'flying',
-    'grass',
-    'poison',
-    'electric',
-    'ground',
-    'psychic',
-    'rock',
-    'ice',
-    'bug',
-    'dragon',
-    'ghost',
-    'dark',
-    'steel',
-    'fairy',
+    { value: 'normal', label: 'Normal', color: '#a8a77a' },
+    { value: 'fire', label: 'Fire', color: '#ee8130' },
+    { value: 'fighting', label: 'Fighting', color: '#c22e28' },
+    { value: 'water', label: 'Water', color: '#6390f0' },
+    { value: 'flying', label: 'Flying', color: '#a98ff3' },
+    { value: 'grass', label: 'Grass', color: '#7ac74c' },
+    { value: 'poison', label: 'Poison', color: '#a33ea1' },
+    { value: 'electric', label: 'Electric', color: '#f7d02c' },
+    { value: 'ground', label: 'Ground', color: '#e2bf65' },
+    { value: 'psychic', label: 'Psychic', color: '#f95587' },
+    { value: 'rock', label: 'Rock', color: '#b6a136' },
+    { value: 'ice', label: 'Ice', color: '#96d9d6' },
+    { value: 'bug', label: 'Bug', color: '#a6b91a' },
+    { value: 'dragon', label: 'Dragon', color: '#6f35fc' },
+    { value: 'ghost', label: 'Ghost', color: '#735797' },
+    { value: 'dark', label: 'Dark', color: '#705746' },
+    { value: 'steel', label: 'Steel', color: '#b7b7ce' },
+    { value: 'fairy', label: 'Fairy', color: '#d685ad' },
   ];
-  return types.map(type => (
+
+  const customStyles = {
+    control: (styles: any) => ({
+      ...styles,
+      backgroundColor: 'white',
+    }),
+    option: (
+      styles: any,
+      { data, isDisabled, isFocused, isSelected }: optionObject
+    ) => {
+      const color = chroma(data.color);
+      return {
+        ...styles,
+        backgroundColor: isDisabled
+          ? null
+          : isSelected
+          ? data.color
+          : isFocused
+          ? color.alpha(0.1).css()
+          : null,
```

Comment:
😂

Reply:
Hey man! I just C'd and P'd from [`react-select` custom styles](https://react-select.com/home#custom-styles).